mdal/frmts/mdal_ugrid.cpp

```diff
@@ -401,3 +403,133 @@ void MDAL::DriverUgrid::parse2VariablesFromAttribute( const std::string &name, c
+
+    // Write globals
+    writeGlobals( );
+
+    // Write variables
+    writeVariables( mesh );
+  }
+  catch ( MDAL_Status error )
+  {
+    if ( status ) *status = ( error );
+  }
+}
+
+
+void MDAL::DriverUgrid::writeVariables( MDAL::Mesh *mesh )
+{
+  // Global dimensions
+  int dimNodeCountId = mNcFile.defineDimension( "nmesh2d_node", mesh->verticesCount() );
+  int dimFaceCountId = mNcFile.defineDimension( "nmesh2d_face", mesh->facesCount() );
+  int dimMaxNodesPerFaceId = mNcFile.defineDimension( "max_nmesh2d_face_nodes",
+                             mesh->faceVerticesMaximumCount() );
+  // Mesh 2D Definition
+  int mesh2dId = mNcFile.defineVar( "mesh2d", NC_INT, 0, nullptr );
+  mNcFile.putAttrStr( mesh2dId, "cf_role", "mesh_topology" );
+  mNcFile.putAttrStr( mesh2dId, "long_name", "Topology data of 2D network" );
+  mNcFile.putAttrInt( mesh2dId, "topology_dimension", 2 );
+  mNcFile.putAttrStr( mesh2dId, "node_coordinates", "mesh2d_node_x mesh2d_node_y" );
+  mNcFile.putAttrStr( mesh2dId, "node_dimension", "nmesh2d_node" );
+  mNcFile.putAttrStr( mesh2dId, "max_face_nodes_dimension", "max_nmesh2d_face_nodes" );
+  mNcFile.putAttrStr( mesh2dId, "edge_node_connectivity", "mesh2d_edge_nodes" );
+  mNcFile.putAttrStr( mesh2dId, "edge_dimension", "nmesh2d_edge" );
+  mNcFile.putAttrStr( mesh2dId, "face_node_connectivity", "mesh2d_face_nodes" );
+  mNcFile.putAttrStr( mesh2dId, "face_dimension", "nmesh2d_face" );
+
+  // Nodes X coordinate
+  int mesh2dNodeXId = mNcFile.defineVar( "mesh2d_node_x", NC_DOUBLE, 1, &dimNodeCountId );
+  mNcFile.putAttrStr( mesh2dNodeXId, "units", "m" );
+  mNcFile.putAttrStr( mesh2dNodeXId, "standard_name", "projection_x_coordinate" );
+  mNcFile.putAttrStr( mesh2dNodeXId, "long_name", "x-coordinate of mesh nodes" );
+  mNcFile.putAttrStr( mesh2dNodeXId, "mesh", "mesh2d" );
+  mNcFile.putAttrStr( mesh2dNodeXId, "location", "node" );
+
+  // Nodes Y coordinate
+  int mesh2dNodeYId = mNcFile.defineVar( "mesh2d_node_y", NC_DOUBLE, 1, &dimNodeCountId );
+  mNcFile.putAttrStr( mesh2dNodeYId, "units", "m" );
+  mNcFile.putAttrStr( mesh2dNodeYId, "standard_name", "projection_y_coordinate" );
+  mNcFile.putAttrStr( mesh2dNodeYId, "long_name", "y-coordinate of mesh nodes" );
+  mNcFile.putAttrStr( mesh2dNodeYId, "mesh", "mesh2d" );
+  mNcFile.putAttrStr( mesh2dNodeYId, "location", "node" );
+
+  // Nodes Z coordinate
+  int mesh2dNodeZId = mNcFile.defineVar( "mesh2d_node_z", NC_DOUBLE, 1, &dimNodeCountId );
+  mNcFile.putAttrStr( mesh2dNodeZId, "mesh", "mesh2d" );
+  mNcFile.putAttrStr( mesh2dNodeZId, "location", "node" );
+  mNcFile.putAttrStr( mesh2dNodeZId, "coordinates", "mesh2d_node_x mesh2d_node_y" );
+  mNcFile.putAttrStr( mesh2dNodeZId, "standard_name", "altitude" );
+  mNcFile.putAttrStr( mesh2dNodeZId, "long_name", "z-coordinate of mesh nodes" );
+  mNcFile.putAttrStr( mesh2dNodeZId, "units", "m" );
+  mNcFile.putAttrStr( mesh2dNodeZId, "grid_mapping", "projected_coordinate_system" );
```

Comment:
you need to store CRS in this array if you aldready defined it, see https://github.com/lutraconsulting/MDAL/blob/master/mdal/mdal_data_model.hpp#L183

but only if it is not-null. (there are diferent options, you probably need to parse back EPSG code if there is one, or store whole proj-string in case it is stored in that way)